tests/utils/exec/rosacli/pr_check.go

```diff
@@ -30,22 +32,37 @@ func GetCommitFoucs() (string, error) {
 		return "", err
 	}
 	fmt.Printf("\nThe last commit is: %s\n", output.String())
-	theStrSlice := strings.Split(output.String(), " ")
-
-	var tcIDs []string
-	reg := regexp.MustCompile(`(id:.*)`)
-	for _, theStr := range theStrSlice {
-		m := reg.FindAllString(theStr, -1)
-		if len(m) > 0 {
-			for _, idStr := range m {
-				idStr = strings.Split(idStr, "id:")[1]
-				ids := strings.Split(idStr, ",")
-				tcIDs = append(tcIDs, ids...)
-			}
-		}
+	return output.String(), err
+}
+
+func GetFocusCaseIDs(commitMessage string) (string, error) {
+	reg := regexp.MustCompile(`ids?:([0-9,\s,]*)`)
+	idsMatched := reg.FindAllStringSubmatch(commitMessage, -1)
+	focus := ""
+	if len(idsMatched) == 0 {
+		return focus, nil
+	}
+	var ids = []string{}
+	for _, matched := range idsMatched {
+		ids = append(ids, common.ParseCommaSeparatedStrings(matched[1])...)
 	}
 
-	focus := strings.Join(tcIDs, "|")
-	_, err = common.CreateFileWithContent(config.Test.TestFocusFile, focus)
+	focus = strings.Join(ids, "|")
+	_, err := common.CreateFileWithContent(config.Test.TestFocusFile, focus)
 	return focus, err
 }
+
+func GetFeatureLabelFilter(commitMessage string) (*Labels, error) {
+	featureFilterMap := map[*Labels]regexp.Regexp{
+		&labels.Feature.Machinepool: *regexp.MustCompile(`[Ff]eature\s?:\s?[mM]achine[\s-]?[pP]ools?`),
```

Comment:
I think we could follow up the same logic on case's, and then do a strings.Contain . The user should get the accurate feature label filter from https://github.com/openshift/rosa/blob/master/tests/ci/labels/features.go

Reply:
I would like code resolve all things instead of keeping rules by personality. With this map, whatever the label changed it will  work all the time.